packages/knip/src/plugins/eleventy/index.ts

```diff
@@ -13,14 +16,31 @@ const isEnabled: IsPluginEnabledCallback = ({ dependencies }) => hasDependency(d
 
 const ENTRY_FILE_PATTERNS = ['.eleventy.js', 'eleventy.config.{js,cjs}'];
 
-const PRODUCTION_ENTRY_FILE_PATTERNS = ['posts/**/*.11tydata.js', '_data/**/*.{js,cjs,mjs}'];
+const PRODUCTION_ENTRY_FILE_PATTERNS = ['**/*.11tydata.js', '_data/**/*.{js,cjs,mjs}']
+  .map(pattern => [`src/${pattern}`, `content/${pattern}`])
+  .flat();
 
 const findEleventyDependencies: GenericPluginCallback = async (configFilePath, options) => {
   const { config } = options;
 
-  return config.entry
-    ? config.entry.map(toProductionEntryPattern)
-    : [...ENTRY_FILE_PATTERNS.map(toEntryPattern), ...PRODUCTION_ENTRY_FILE_PATTERNS.map(toProductionEntryPattern)];
+  let localConfig = (await load(configFilePath)) as
+    | Partial<EleventyConfig>
+    | ((arg: DummyEleventyConfig) => Promise<Partial<EleventyConfig>>);
+  if (!localConfig) return PRODUCTION_ENTRY_FILE_PATTERNS.map(toProductionEntryPattern);
+  if (typeof localConfig === 'function') localConfig = await localConfig(new DummyEleventyConfig());
+
+  const inputDir = localConfig?.dir?.input || defaultEleventyConfig.dir.input;
+  const dataDir = localConfig?.dir?.data || defaultEleventyConfig.dir.data;
+  const templateFormats = localConfig.templateFormats || defaultEleventyConfig.templateFormats;
+
+  return (
+    config?.entry ?? [
+      join(inputDir, dataDir, '**/*.js'),
+      join(inputDir, `**/*.{${typeof templateFormats === 'string' ? templateFormats : templateFormats.join(',')}}`),
+      join(inputDir, '**/*.11tydata.js'),
+    ] ??
+    ENTRY_FILE_PATTERNS
```

Comment:
I think this will never resolve to `ENTRY_FILE_PATTERNS` as the last `??` is behind an array. Looks like we can remove it?